pkg/proxy/winkernel/proxier.go

```diff
@@ -1157,6 +1158,26 @@ func (proxier *Proxier) syncProxyRules() {
 		return
 	}
 	if strings.EqualFold(proxier.network.networkType, NETWORK_TYPE_OVERLAY) {
+		/* Source VIP is required for overlay network. KubeProxy will determine the Source Vip as follows:
+		1. Use the Source Vip in the KubeProxy winkernel configuration options
+		2. If the Source Vip is not provided in the winkernel configuration, check if an HNS endpoint exists with the well-known name
+		   It is the responsibility of the CNI to ensure the Source Vip is created using the well known-documented name
+		3. If the expected Source Vip endpoint name does not exist on the host, KubeProxy will attempt to create the Source Vip
```

Comment:
I think step 3 will fail if we fail to find the ip in step 2 or 3 since we will have an empty ip address.  I was reading up on this again and I think we can't create the IP address anyways due to the fact that it is not stored in the CNI's IPAM (https://github.com/kubernetes/kubernetes/issues/100962#issuecomment-821274716).  I actuall think we need to eventuall drop the code that creates the sourcevip in the lines below.

Reply:
I agree with @jsturtevant , it would be better for kube-proxy to not create the source-vip. However, given that this code already exists, this should not be PR blocking.

Reply:
I actually ran a test prior to opening the PR and it failed. Previously there seemed to never be a code path to reach this method because if the source VIP was not provided as a kube proxy configuration it would err and return. I was hesitant to change the existing code. Thre is a related suggestion from James to move the logic to look it up the source VIP code earlier so that we can fail fast. So I think resolving that comment will also resolve this.

Reply:
>  I actuall think we need to eventuall drop the code that creates the sourcevip in the lines below

@jsturtevant I agree. Shall I drop the code with this change as well? Am hesitant to potentially add a breaking change but I can add a TODO about deprecating that section of the code.

Reply:
TODO in this PR is good and let's drop the code as different PR